Sources/XcodeGen/Logger.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+import Rainbow
+
+/// Mechanism for logging informational, success, and fatal messages.
+struct Logger {
+
+    // MARK: - Properties
+
+    /// Should informational and success messages be suppressed
+    let isQuiet: Bool
+
+    /// Should the output be colored
+    let isColored: Bool
+
+    // MARK: - Initializers
+
+    /// Initialize a logger
+    ///
+    /// - parameter isQuiet: should informational and success messages be suppressed
+    /// - parameter isColored: should the output be colored
+    init(isQuiet: Bool = false, isColored: Bool = true) {
+        self.isQuiet = isQuiet
+        self.isColored = isColored
+    }
+
+    // MARK: - Logging
+
+    /// Log a fatal message and exit with status code 1
```

Comment:
This is why I stay away from doc comments as they always get out of date. 😄  In this case, it mentions status code here (and in success). 
If this doesn't exit anymore, we could also rename this to `error`.

Also if we wanted to remove all the duplicate `exit(1)` in the main file, a simple helper function there called `fatalError(_ message: String)` could log an error and then exit

Reply:
Sorry for not seeing this before :)

Reply:
`fatalError(_ message: String)` good use case for `-> Never` @soffes 😜